Gems/EMotionFX/Code/Source/Integration/Components/AnimGraphComponent.cpp

```diff
@@ -433,6 +433,24 @@ namespace EMotionFX
             return 0;
         }
 
+        void AnimGraphComponent::SetActorThreadIndex(AZ::u32 threadIndex)
+        {
+            if (m_actorInstance)
+            {
+                m_actorInstance->SetThreadIndex(threadIndex);
+            }
+        }
+
+        AZ::u32 AnimGraphComponent::GetActorThreadIndex() const
+        {
+            if (m_actorInstance)
+            {
+                return m_actorInstance->GetThreadIndex();
+            }
+
+            return 0;
```

Comment:
question: Is 0 ever a valid thread index? Should we instead return a nullopt here or some invalid value?

Reply:
Seems that's the default value in : 
`static ActorInstance* Create(Actor* actor, AZ::Entity* entity = nullptr, uint32 threadIndex = 0);`

It's not actually used anywhere, added more for it to be paired with the setter

Reply:
Okay cool understood, thanks for confirming (maybe could include a comment to say `// default thread index` perhaps?)

Reply:
Some question:
1. Does this actually work setting the threadIndex directly hence bypasses the multiThreadScheduler in emfx? Sounds a little too good to be true.. Perhaps the networking system in MP is calling those in a different multi threaded job system?
2. Shouldn't this be part of actor component instead of anim graph component, since it's directly changing the actor instance?
3. We might want to have this on the editor component as well (actor component vs editor actor component)

Reply:
1. MultiThreadScheduler isn't used here since in multiplayer each actor is updated separately via `AnimGraphComponent::UpdateActorExternal`. 

2. Similarly to how `AnimGraphComponent::UpdateActorExternal` is called and changing the actor instance, we do the same with thread index. 
If you feel `UpdateActorExternal` should be removed from `AnimGraphComponent` to something like `ActorComponent::UpdateExternal`, then of course we can move `SetThreadIndex` too. 
Or in fact maybe just pass the thread index as an argument to `UpdateActorExternal`?

3. if we UpdateActorExternal in the Editor, could have it too. Otherwise if there's no manual update, we can simply use MultiThreadScheduler which does set thread index internally.

Reply:
Regarding 2 I definitely think they should be part of the actor component.